src/CommandSystem/JobListener.h

```diff
@@ -20,7 +20,7 @@ namespace Command {
 using FileWriter::Msg;
 
 /// \brief Check for new jobs on a topic.
-class JobListener : private CommandListener {
+class JobListener : public CommandListener {
 public:
   /// \brief The constructor will not automatically connect to the Kafka broker.
   JobListener(uri::URI JobPoolUri, Kafka::BrokerSettings Settings);
```

Comment:
Currently `enable.auto.commit` is set to false in `BrokerSettings` as we didn't need the feature. If it is set to true then the consumer automatically commits its offset, alternatively we can explicitly call a function to commit the offset.

> due to the possibility of processing old and outdated commands.

I don't think there is such a possibility, unless a filewriter spontaneously dies, in which case we might want the job to be reprocessed anyway.
If we don't use this feature then we must ensure that there is an idle filewriter available at all times, otherwise the pool can miss jobs.

Reply:
After considering this a bit more, there are other checks for ensuring that we are not working on outdated commands so doing this should have a very minor effect on which commands are handled. I have thus enabled auto commit.